packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -0,0 +1,157 @@
+
+Example:
+
+Suppose a table had 3 columns with the following configuration: `currentFractionalWidth = 1`, `currentFixedWidth = 100`, and `currentFractionalWidth = 2`. This would result in "`grid-templateColumns: 1fr 100px 2fr;`".
+
+#### **Managing interactive resize**
+
+In order to facilitate proper resizing of columns that are leveraging  `currentFractionalWidth` over `currentFixedWidth` (by way of a column mixin), it can be useful to set the `currentFixedWidth` to the appropriate calculated value (requiring knowledge of the current row width) on mouse down, and then updating `currentFixedWidth` according to the mouse delta during the mouse move, and then resetting the columns that should be using `currentFractionalWidth` to the appropriate new calculated value (based on the final `currentFixedWidth` value along with the current row width).
+
+Because we will allow a horizontal scrollbar once the right-most column reaches its minimum size, there is an implicit need for us to also size the container for the rows and header to the actual pixel sizes the columns resolve to (i.e. the container width becomes larger than the actual table width). Additionally, on a table resize, the row size should be updated by the same delta, to maintain the behavior of columns using `currentFractionalWidth` to be proportionlly sized as expected.
+
+#### **Future interactive sizing behaviors**
+
+Currently, we only have a need for an interactive resize to behave as outlined in the 'Expected Behaviors' section, but it is recognized that another behavior we will ultimately want to support is a "push" resize mode, where as a user drags a divider to the right, all columns are simply pushed to the right, maintaining their current pixel size. 
+
+To help facilitate this, it may be helpful to approach the initial implementation by encapsulating the column resize management within a separate class from the `Table` and have it implement a common interface that would allow a separate implementation to provide differing behavior.
+
+## Alternative Implementations / Designs
+
+TanStack offers the ability to maintain column sizing state as well as APIs to manage interactive sizing ([`getResizeHandler`](https://tanstack.com/table/v8/docs/api/features/column-sizing#getresizehandler)). By plugging into `getResizeHandler` TanStack can provide all necessary column size state either as the column is being sized (allowing immediate sizing while dragging), or at the end of an interactive operation (columns update size on mouse up for example).
+
+TanStack expects size values to be provided as [pixel values](https://tanstack.com/table/v8/docs/api/features/column-sizing#size). As such, we wouldn't be able to leverage TanStack's APIs in a way to achieve our initial desired behavior. However, it's possible that it would be beneficial to attempt to upstream changes to TanStack that would allow us to leverage it, but I would suggest that we revisit this possibility at a later time.
```

Comment:
My understanding is that not only does TanStack require pixel columns but it also doesn't have a resize mode that aligns with what we're trying to achieve. Is that correct? I ask because I think the answer has a notable impact on the effort required to upstream our resize behavior.

Reply:
Correct, they do not have a resize mode that aligns with our expected behavior. I would expect upstreaming changes that would allows us to use TanStack to likely be a significant effort.